### ISSUE TITLE
Mask for counts in create_area_weights.py counts all records in masked group, not just the records with nonzero variable values

### ISSUE THREAD

**Issue by donboyd5** [PSLmodels/tax-microdata-benchmarking#246]
We want to be able to target the number of tax units that have nonzero values for a given variable in a given bin, in addition to targeting all units in a given bin.

Count targets are determined when row.count in a targets file is 1, triggering the code in the `else` condition in line 238 in `create_area_weights.py` below:

https://github.com/PSLmodels/tax-microdata-benchmarking/blob/342578946a245f5a01985250756f56646cf82d61/tmd/areas/create_area_weights.py#L235-L238

(In the code above, numobs is always the number or rows in the dataframe vardf, determined in line 201.)

This is later masked to determine the scope, filing status, and agi bin for which we want a count. However, it is not masked by whether the targeted variable has a nonzero value. Thus, this counts all records in the group, after masking, not just the records with nonzero variable values. For example, if the row has a target for the count of tax units with capital gains for a particular variable with a given agi bin, filing status, and scope the code above will count ALL records in that subset not just those with nonzero capital gains. This is not what we want.

One solution would seem to be amending line 238 to read as:

`unmasked_varray = np.ones(numobs, dtype=float) * (vardf[row.varname].astype(float) != 0.0)`

Another solution might be to create a new mask for whether a variable is nonzero and to apply that on top of the other masks.

Either approach will break something else we care about. In addition to targeting counts for certain variables when they are nonzero, we want to be able to target total counts such as the number of filers. This will be larger than the number of returns that has nonzero AGI. It might be that we can simply target counts for `varname==data_source` since filers will always have `data_source==1` . So we probably could deal with this by using varname data_source when we want to target total filers.

**Comment by martinholmer** [comment]
@donboyd5 said in issue #246:
> We want to be able to target the number of tax units that have nonzero values for a given variable in a given bin, in addition to targeting all units in a given bin.

Why did you not say this two months ago when we designed the targets file format?

**Comment by donboyd5** [comment]
I did say it, in a way I understood, but it turns out you and I were speaking a different language. 

In the previous project, we had constructed variable-specific count targets when we did the national targeting; it was an important part of our targeting. In this project I had talked at various points about the importance of targeting numbers of returns with capital gains income, or Social Security income, or whatever (both of which we did for the nation). I assumed in those conversations that you knew I was talking about variable-specific counts. And it was consistent with what the target files allowed - we specify a variable, and we specify a count indicator, suggesting that one can choose to target either a count or any amoung for any variable, which is exactly what I wanted to be able to do. After I found cases when we weren't hitting targets the way I expected, and worked through the code, I realized the code wasn't doing what I expected it to do.

I don't think it's a big deal in the short run. It would be great if we can add the capability for variable-specific targets later in this project but I don't think it should be at the top of our priority list. We can figure out where it belongs on the priority list, or whether it can stay on the list, next time we talk on the phone.
